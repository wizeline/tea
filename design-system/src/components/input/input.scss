wz-input {
  .error .input-container {
    border-color: var(--negative-base);
    &:focus-within {
      border-color: var(--negative-base);
    }
  }

  wz-error {
    display: block;
    height: 16px;
  }

  .input-container {
    display: flex;
    align-items: center;
    border-radius: 3px;
    background-color: var(--ink-clear);
    border: solid 1px;
    height: 40px;
    overflow: hidden;
    border-color: var(--ink-light);

    &:focus-within {
      border-color: var(--primary-base);
    }

    input {
      padding: 8px 16px;
      outline: none !important;
      border: none;
      font-size: 14px;
      width: 100%;

      &:disabled {
        color: var(--neutral-dark);
      }
    }
  }
}
