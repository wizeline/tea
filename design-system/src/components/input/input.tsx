import { Component, Prop } from '@stencil/core';

@Component({
  tag: 'wz-input',
  styleUrl: 'input.scss',
  shadow: false,
})
export class Input {
  @Prop({ mutable: true })
  value: string;
  @Prop()
  name: string;
  @Prop()
  type: string = 'text';
  @Prop()
  disabled: boolean = false;
  @Prop()
  placeholder: string;
  @Prop()
  error: string;
  @Prop()
  class: string;

  render() {
    const commonProps = {
      disabled: this.disabled,
      name: this.name,
      placeholder: this.placeholder,
    };

    const classes = [this.class, 'input-container'].join(' ');

    return (
      <wz-form-error error={this.error}>
        <div class={classes}>
          <input value={this.value} type={this.type} {...commonProps} />
        </div>
      </wz-form-error>
    );
  }
}
