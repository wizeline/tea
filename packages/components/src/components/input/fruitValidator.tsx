import { Validator } from './validator';

export const FruitValidator: Validator<string> = {
  validate: (value: string) => {
    let fruits = ['banana', 'apple', 'cherry'];
    return fruits.find(a => a === value) ? true : false;
  },
  errorMessage: 'You must enter a valid fruit name'
}
