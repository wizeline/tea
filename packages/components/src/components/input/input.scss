wz-input {
  .input-container {
    border-radius: 3px;
    background-color: var(--ink-clear);
    border: solid 1px;
    height: 40px;
    padding: 8px;
    border-color:  var(--neutral-base);

    &:focus-within {
      border-color:  var(--primary-light);
    }

    input {
      outline: none !important;
      border: none;
      font-size: 14px;
      width: 100%;
    }
  }

}
