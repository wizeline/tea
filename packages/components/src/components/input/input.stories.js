import { storiesOf } from '@storybook/html';

storiesOf('Input', module).add(
  'default',
  () => '<wz-input value="Default Input"></wz-input>',
);
