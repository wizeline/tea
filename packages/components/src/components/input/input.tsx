import { Component, Prop, Event, EventEmitter } from '@stencil/core';


@Component({
  tag: 'wz-input',
  styleUrl: 'input.scss',
  shadow: false,
})
export class Input {

  @Prop({ mutable: true }) value: string;

  @Event() changed: EventEmitter<string>;

  handleChange(ev) { 
    this.value = ev.target ? ev.target.value : null;
    this.changed.emit(this.value);
  }

  render() {    
    return (
      <div class="input-container">
        <input value={this.value} onInput={this.handleChange} />
      </div>
    );
  }
}
