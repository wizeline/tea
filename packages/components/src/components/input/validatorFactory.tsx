import { FruitValidator } from './fruitValidator';
import { Validator } from './validator';

export enum ValidatorsName {
  fruit = 'fruit',
};

export function validatorFactory(name: string): Validator<any> {
  switch(name) {
    case (ValidatorsName.fruit):
      return FruitValidator;
    default:
      return defaultValidator;
  }
}

const defaultValidator: Validator<any> = {
  validate: (_x: any) => true
}
