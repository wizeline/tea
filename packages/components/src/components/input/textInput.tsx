import { Component, Prop, Event, EventEmitter } from '@stencil/core';
import { Validator } from './validator';
import { validatorFactory } from './validatorFactory';

@Component({
  tag: 'wz-validator',
  styleUrl: 'input.scss',
  shadow: false,
})
export class textInput {
  @Prop({ mutable: true })
  value: string;
  @Prop() validator: string;
  @Event() changed: EventEmitter<string>;

  _validator: Validator<string> = defaultValidator;

  componentWillLoad() {
    this._validator = validatorFactory(this.validator);
  }

  componentWillUpdate() {
    this._validator = validatorFactory(this.validator);
  }

  handleChange(ev) {
    this.value = ev.target ? ev.target.value : null;
    this.changed.emit(this.value);
  }

  render() {
    console.log(this._validator, this._validator.validate(this.value));

    return (
      <div>
        <div class="input-container">
          <input value={this.value} onInput={ev => this.handleChange(ev)} />
        </div>
        {!this._validator.validate(this.value) ? (
          <span class="validation-error">{this._validator.errorMessage}</span>
        ) : null}
      </div>
    );
  }
}

const defaultValidator: Validator<any> = {
  validate: (_x: any) => true,
};
