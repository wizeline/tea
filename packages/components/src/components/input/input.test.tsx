import { Input } from './input';

import { TestWindow } from '@stencil/core/testing';

describe('wz-input', () => {
  // let element;
  it('renders', async () => {
    const window = new TestWindow();
    const element = await window.load({
      components: [Input],
      html: '<wz-input value=\'Hello, my name is Marty McFly\'></wz-input>',
    });

    console.log(element, '-----------')

    expect(element.textContent).toEqual('Hello, my name is Marty McFly');
  });

  // beforeEach(async () => {
  //   const window = new TestWindow();
  //   element = await window.load({ components: [Input], html: '<wz-input value="test"></wz-input><p>test</p>' });
  //   console.log('------await')
  //   // element = await render();
  // });
  // describe('handleToggle()', () => {
  //   it('emits changed event', () => {
  // //     const cmp = new Input;
  //     console.log(window.document.getElementsByTagName('wz-input').length, '-----00-------');
  //     // const spy = jest.spyOn(element.changed, 'emit');
  //     console.log(element, '--------2---');
  //     // element.handleChange('test');
  //     // expect(spy).toHaveBeenCalled();
  //   });
  // });
});



// it('should be the new way', async () => {
//   const window = new TestWindow();
//   const element = await window.load({
//     components: [MyComponent],
//     html: '<my-cmp first="Marty" last-name="McFly"></my-cmp>',
//   });
//   expect(element.textContent).toEqual('Hello, my name is Marty McFly');

//   element.first = 'George';
//   await window.flush();

//   expect(element.textContent).toEqual('Hello, my name is George McFly');
// });
