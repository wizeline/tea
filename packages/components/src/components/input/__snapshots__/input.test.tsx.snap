// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wz-input renders disabled 1`] = `
<wz-input
  class="hydrated"
  disabled="false"
>
  
  <div
    class="input-container"
  >
    <input
      type="text"
    />
  </div>
</wz-input>
`;

exports[`wz-input renders value 1`] = `
<wz-input
  class="hydrated"
  value="Hello, my name is Marty McFly"
>
  
  <div
    class="input-container"
  >
    <input
      type="text"
    />
  </div>
</wz-input>
`;
